presto-main/src/main/java/com/facebook/presto/cost/ComparisonStatsCalculator.java

```diff
@@ -0,0 +1,188 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.cost;
+
+import com.facebook.presto.sql.planner.Symbol;
+import com.facebook.presto.sql.tree.ComparisonExpressionType;
+
+import static com.facebook.presto.cost.FilterStatsCalculator.filterStatsForUnknownExpression;
+import static com.facebook.presto.cost.SymbolStatsEstimate.buildFrom;
+import static java.lang.Double.NEGATIVE_INFINITY;
+import static java.lang.Double.NaN;
+import static java.lang.Double.POSITIVE_INFINITY;
+import static java.lang.Double.isNaN;
+import static java.lang.Math.max;
+
+public class ComparisonStatsCalculator
+{
+    private final PlanNodeStatsEstimate inputStatistics;
+
+    ComparisonStatsCalculator(PlanNodeStatsEstimate inputStatistics)
+    {
+        this.inputStatistics = inputStatistics;
+    }
+
+    public PlanNodeStatsEstimate comparisonSymbolToLiteralStats(Symbol symbol, double doubleLiteral, ComparisonExpressionType type)
+    {
+        switch (type) {
+            case EQUAL:
+                return symbolToLiteralEquality(symbol, doubleLiteral);
+            case NOT_EQUAL:
+                return symbolToLiteralNonEquality(symbol, doubleLiteral);
+            case LESS_THAN:
+            case LESS_THAN_OR_EQUAL:
+                return symbolToLiteralLessThan(symbol, doubleLiteral);
+            case GREATER_THAN:
+            case GREATER_THAN_OR_EQUAL:
+                return symbolToLiteralGreaterThan(symbol, doubleLiteral);
+            case IS_DISTINCT_FROM:
+                break;
+        }
+
+        return filterStatsForUnknownExpression(inputStatistics);
+    }
+
+    private PlanNodeStatsEstimate symbolToLiteralGreaterThan(Symbol symbol, double literal)
+    {
+        SymbolStatsEstimate symbolStats = inputStatistics.getSymbolStatistics(symbol);
+
+        StatisticRange range = new StatisticRange(symbolStats.getLowValue(), symbolStats.getHighValue(), symbolStats.getDistinctValuesCount());
+        StatisticRange intersectRange = range.intersect(new StatisticRange(literal, POSITIVE_INFINITY, NaN));
+
+        double filterFactor = range.overlapPercentWith(intersectRange);
+        SymbolStatsEstimate symbolNewEstimate =
+                SymbolStatsEstimate.builder()
+                        .setAverageRowSize(symbolStats.getAverageRowSize())
+                        .setDistinctValuesCount(intersectRange.getDistinctValuesCount())
+                        .setHighValue(intersectRange.getHigh())
+                        .setLowValue(intersectRange.getLow())
+                        .setNullsFraction(0.0).build();
```

Comment:
why?
Out of `3601` usages of `.build()` method in presto codebase only `1454` (`~40%`) are called in new line.